### ISSUE TITLE
EXWM GNOME Flashback:  Window border of  `gnome-screenshot` is not fully drawn

### ISSUE THREAD

**Issue by emacksnotes** [ch11ng/exwm#634]
### EXWM GNOME Flashback:  Window border of  `gnome-screenshot` is not fully drawn

**`gnome-screenshot` window has half-baked border on left and right sides, and no border on the bottom side**
![Screenshot from 2019-08-27 18-48-45](https://user-images.githubusercontent.com/42199775/63774747-8a0d8880-c8fb-11e9-99fb-c530ec1c83fc.png)

```elisp
(custom-set-variables
 '(exwm-debug t)
 '(exwm-floating-border-color "red")
 '(exwm-floating-border-width 2)
  '(exwm-manage-force-tiling nil))
```

I am on this commit

```
commit 5b9f4b0851d27e8132a930dfbfec0b94099a4c9f (HEAD -> externals/exwm, origin/externals/exwm)
Author: Chris Feng <chris.w.feng@gmail.com>
Date:   Sun Aug 18 00:00:00 2019 +0000
```

**Comment by emacksnotes** [comment]
> EXWM GNOME Flashback: Window border of gnome-screenshot is not fully drawn

Found out what the problem is.  

I have `menu-bar-mode` on.  Look at the `gnome-screenshot` window in my bug report, and you will see that it has a `menu-bar`.


### Note

I built custom menus for most packages I use.  Look at the screenshot again, and you will see that there are menus that are specific to my setup.  All my posts on my blog [Emacs Notes – GNU Emacs, for Ordinary Folks](https://emacsnotes.wordpress.com/) use menus.   I found that `Buffers` menu is particularly useful in `char-mode` buffers.  In those buffers, I cannot use the regular Emacs keystrokes for windows and buffers, and I can use invoke the buffer, window and frame commands using mouse operations on the menu.  Now that I have "migrated" buffer, window and frame commands to EXWM keystrokes (see https://github.com/ch11ng/exwm/issues/635#issue-486211726), I can safely turn off the menu bar for frames such as the one created by `gnome-screenshot`.

**Comment by ch11ng** [comment]
Using `menu-bar-mode` can be a confusing at the first sight that it looks like the menu bar of the managed X window.  Anyway it's a bug and I'll push a fix.

**Comment by ch11ng** [comment]
I've pushed the fix: ccc4cce0e0df426f4eccb81479c4a773c488155d .

**Comment by emacksnotes** [comment]
> EXWM doesn't account for menu-bar-mode

>> I've pushed the fix: ccc4cce .

The fix works, but only partially.  The window border is OK, but the color is NOT OK.

I am attaching three screeshots as I progressively turn of  `tool-bar-mode` and `menu-bar-mode` sequentially.

## After the fix

![Screenshot from 2019-09-09 07-39-12](https://user-images.githubusercontent.com/42199775/64499143-71c34380-d2d5-11e9-96a6-e43318883992.png)
![Screenshot from 2019-09-09 07-38-54](https://user-images.githubusercontent.com/42199775/64499145-71c34380-d2d5-11e9-96c1-b085b043fdcc.png)
![Screenshot from 2019-09-09 07-38-33](https://user-images.githubusercontent.com/42199775/64499148-725bda00-d2d5-11e9-9c79-0a0c29013cee.png)

### Before the fix

**`gnome-screenshot` window has half-baked border on left and right sides, and no border on the bottom side**
![Screenshot from 2019-08-27 18-48-45](https://user-images.githubusercontent.com/42199775/63774747-8a0d8880-c8fb-11e9-99fb-c530ec1c83fc.png)

**Comment by emacksnotes** [comment]
> > EXWM doesn't account for menu-bar-mode
> 
> > > I've pushed the fix: [ccc4cce](https://github.com/ch11ng/exwm/commit/ccc4cce0e0df426f4eccb81479c4a773c488155d) .
> 
> The fix works, but only partially. The window border is OK, but the color is NOT OK.

Your fix works fully.

When I restarted the laptop today, I noticed that the floating window's border and width are honored.  

The problem is that both `exwm-floating-border-color` and `exwm-floating-border-width` do not define any setters.  I used the custom interface, and my expectation was that settings would  take effect immediately. 

```elisp
(defcustom exwm-floating-border-color "navy"
  "Border color of floating windows."
  :type 'color)

(defcustom exwm-floating-border-width 1
  "Border width of floating windows."
  :type 'integer)

```

```elisp
(custom-set-variables
 '(exwm-debug t)
 '(exwm-floating-border-color "red")
 '(exwm-floating-border-width 20)
 '(exwm-manage-force-tiling nil))
```

**Comment by ch11ng** [comment]
The other issue is fixed with 48b15e25ad9009b1135db6344de156ef4154758f.
